homeassistant/components/wemo/entity.py

```diff
@@ -136,6 +138,24 @@ def is_on(self) -> bool:
         """Return true if the state is on. Standby is on."""
         return self._state
 
+    @property
+    def should_poll(self) -> bool:
+        """Return True if the the device requires local polling, False otherwise.
+
+        Polling can be disabled if three conditions are met:
+        1. The device has polled to get the initial state (self._has_polled).
+        2. The polling was successful and the device is in a healthy state
+           (self.available).
+        3. The pywemo subscription registry reports that there is an active
+           subscription and the subscription has been confirmed by receiving an
+           initial event. This confirms that device push notifications are
+           working correctly (registry.is_subscribed - this method is async safe).
+        """
```

Comment:
Does this need to be reset when the device is unplugged and plugged back in?

Reply:
That's a nice catch and a good question. 

What should happen is that the device will respond to updates from Home Assistant as soon as it is back on the network, but state updates from when the device is controlled outside of Home Assistant won't be seen right away - for up to 5 minutes - and the current device state won't be picked up till that point either. There is a refresh of the event subscription that happens every 5 minutes. And when that happens, the UPnP spec states that the device will send it's current state as soon as the new subscription is started. So that _should_ pick up any state changes that happened due to the device being unplugged without needing to poll.

In testing this though, I know that there are [two WeMo devices that don't follow the UPnP spec](https://github.com/pavoni/pywemo/pull/236#issuecomment-776115015) and won't send that initial event to update the current state. And so the state of the device will be inconsistent from Home Assistant if the device was On when it was unplugged (WeMo devices default to Off when they are plugged in).

Let's take a pause on this PR. There is a very simple change I can make to pywemo that will solve this edge case. It's easy to detect the case when a new subscription is started vs a renewal. When the device is plugged back in that'll behave as-if a new subscription is started. I'll just reset the boolean that tracks that the subscription has been confirmed by receiving the initial event at that point. That'll fix this corner case. For devices that follow the UPnP spec, that initial event will update the state in Home Assistant. And for devices that don't follow the spec, they'll start polling again.

There will still be a period where the device state will be inconsistent, for up to 5 minutes after it is plugged back in. I'm assuming that is okay since unplugging the device and plugging it back in should be a rare event. But please correct me if my assumption is wrong here.

Reply:
It seems reasonable to have a recovery period after plugging back in the device 👍

Reply:
This should work now (#46825) with the 5 minute recovery period.

I manually verified everything in the description of https://github.com/pavoni/pywemo/pull/247

Reply:
Note that `should_poll` only has an effect for `Entity` built in polling mechanism when adding the entity. Ie the registration of the update on interval only happens during entity addition, not on every state update.

Reply:
Was addressed in https://github.com/home-assistant/core/pull/47191